src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -57,26 +59,24 @@ class HuiShoppingListCard extends LitElement implements LovelaceCard {
     this._config = config;
     this._uncheckedItems = [];
     this._checkedItems = [];
-    this._fetchData();
   }
 
   public connectedCallback(): void {
     super.connectedCallback();
-
-    if (this.hass) {
-      this._unsubEvents = this.hass.connection.subscribeEvents(
-        () => this._fetchData(),
-        "shopping_list_updated"
-      );
-      this._fetchData();
+    if (!this.hass) {
```

Comment:
I wonder if we can use the subscribeMixin. We can have the subscribe callback do the fetching before returning the callback?

Reply:
Yes that would be possible, with some adjustments to the `subscribeMixin` to deal with promises.